File: algorithm/108. Convert Sorted Array to Binary Search Tree/108.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    public TreeNode sortedArrayToBST(int[] nums) {
        return helper(nums, 0 ,nums.length -1);
    }
    private TreeNode helper(int[] nums, int start, int end)
    {
        if(start > end)    return null;
        int mid = (start + end)/2;
        TreeNode tem = new TreeNode(nums[mid]);
        tem.left = helper(nums, start, mid -1);
        tem.right = helper(nums, mid +1, end);
        return tem;
    }
}

File: algorithm/108. Convert Sorted Array to Binary Search Tree/108.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
    def sortedArrayToBST(self, nums):
        """
        :type nums: List[int]
        :rtype: TreeNode
        """
        def helper(nums,start,end):
            if end <= start:
                return None
            mid=(start+end)/2
            root=TreeNode(nums[mid])
            root.left=helper(nums,start,mid)
            root.right=helper(nums,mid+1,end)
            return root
        return helper(nums,0,len(nums))
